viz.Rmd

```diff
@@ -238,48 +413,68 @@ islands_bar <- ggplot(islands_top12, aes(x = size, y = landmass)) +
+`fct_reorder` orders by that column in ascending order by default, 
+but this can be changed to sorting in descending order 
+by setting  `.desc = TRUE`.
+We do this here so that the largest bar will closet to the axis line,
+as that is more visually appealing.
+
+To label the x and y axes, we shift from using `xlab` and `ylab`, 
+to specifying these within the `labs` function. 
+We do this here because we would also like to change the legend label.
+The default label is the column name of the column being mapped to `fill`,
+however `landmass_type` is not very human readable or visually pleasing.
+Thus we use the `fill` argument inside `labs` to change that to "Type".
+Finally, we again \index{ggplot!reorder} use the `theme` function 
+to change the font size:
+
+```{r 03-data-islands-bar-4, warning = FALSE, message = FALSE, fig.width = 8, fig.cap = "Bar plot of size for Earth's largest 12 landmasses coloured by whether its a continent with clearer axes and labels"}
+islands_bar <- ggplot(islands_top12, 
+                      aes(x = size,
+                          y = fct_reorder(landmass, size, .desc = TRUE), 
+                          fill = landmass_type)) +
   geom_bar(stat = "identity") +
   labs(x = "Size (1000 square mi)", y = "Landmass",  fill = "Type") +
-  theme(text = element_text(size = 18), legend.position = c(0.75, 0.45))
+  theme(text = element_text(size = 16))
 islands_bar
 ```
 
 The plot in Figure \@ref(fig:03-data-islands-bar-4) is now a very effective visualization for answering our original questions. Landmasses are organized by their size,
 and continents are coloured differently than other landmasses, making it quite clear that continents are the largest seven landmasses.
 
 
-### The Old Faithful eruption/waiting time data set
-The `faithful` data set \index{Old Faithful} contains measurements of the waiting time between eruptions and the subsequent eruption duration (in minutes). The `faithful` data set is available in base R under the name `faithful` so it does not need to be loaded.
+### The Old Faithful eruption/waiting time data set with scatter plots
+The `faithful` data set \index{Old Faithful} contains measurements 
+of the waiting time between eruptions 
+and the subsequent eruption duration (in minutes). 
+The `faithful` data set is available in base R as a data frame,
+so it does not need to be loaded.
+We convert it to a tibble to take advantage of the nicer print output 
+these specialized data frames provide.
 
 **Question:** \index{question!visualization} Is there a relationship between the waiting time before an eruption to the duration of the eruption? 
 
-```{r 03-data-faithful-tibble, echo = F, warning=FALSE, message=FALSE}
-# old faithful eruption time / wait time data convert to tibble
-faithful <- as.tibble(faithful)
-```
-
 ```{r 03-data-faithful, warning=FALSE, message=FALSE}
 # old faithful eruption time / wait time data
+faithful <- as_tibble(faithful)
 faithful
 ```
 
```

Comment:
I think what is trying to be said is that they do not have a natural order - like time does.

Reply:
More formally, the difference is that the time variable is the CO2 data is not independent, whereas here both variables are are.

Reply:
@leem44 @trevorcampbell I have changed it to this to make it clearer (I want to avoid the term independent, as we have not introduced it, nor do we elsewhere in the book):

```
Here again, we investigate the relationship between two quantitative variables 
(waiting time and eruption time). 
But if you look at the output of the data frame, 
you'll notice that unlike time in the Mauna Loa CO$_{\text{2}}$ data set,
neither of the variables here have a natural order to them.
So, in this case, let's start again with a scatter plot:
```